src/app/functions/server/onerep.ts

```diff
@@ -296,3 +296,27 @@ export async function getScanDetails(
   }
   return response.json() as Promise<Scan>;
 }
+
+export async function getAllScanResults(
+  profileId: number
+): Promise<ScanResult[]> {
+  const scanListFull = [];
+  const firstPage = await listScanResults(profileId, {
+    per_page: 100,
+  });
+  // Results are paginated, use per_page maximum and collect all pages into one result.
+  if (firstPage.meta.last_page > 1) {
+    let currentPage = 2;
+    while (currentPage <= firstPage.meta.last_page) {
+      const nextPage = await listScanResults(profileId, {
+        per_page: 100,
+      });
+      currentPage++;
+      nextPage.data.forEach((element: object) => scanListFull.push(element));
+    }
+  } else {
+    scanListFull.push(firstPage.data);
+  }
+
+  return scanListFull[0];
```

Comment:
what does this `[0]` do?
Looks like we're looping while `currentPage <= firstPage.meta.last_page` and pushing results into `scanListFill[]` Array. But then we might only be returning the first 100 paged results anyways. Should this be `return scanListFull;` or `return scanListFull.flat();`?

Reply:
I just copy/pasted this from the code I landed... I was hoping the webhook had the payload, but looks like it just lets us know the scan is ready and we still need to do the multi-page thing.

I'll go over this code again since it looks like we need to keep it, was hoping it'd only be for local testing.

Reply:
Hm yeah I need to test this (I guess I can reduce the `per_page` setting to do this), but I think you're right - we want to flatten out the array, not just pick the first one.